src/Compiler/Microsoft.CodeAnalysis.Razor.Compiler/src/Language/TagHelperBinder.cs

```diff
@@ -17,28 +17,41 @@ internal sealed class TagHelperBinder
     private readonly ImmutableArray<TagHelperDescriptor> _catchAllDescriptors;
     private readonly Dictionary<string, ImmutableArray<TagHelperDescriptor>> _tagNameToDescriptorsMap;
 
-    public string? TagHelperPrefix { get; }
-    public ImmutableArray<TagHelperDescriptor> TagHelpers { get; }
+    public string? TagNamePrefix { get; }
+    public ImmutableArray<TagHelperDescriptor> Descriptors { get; }
+
+    private readonly ReadOnlyMemory<char> _tagNamePrefix;
```

Comment:
Why do we store this separately from `TagNamePrefix`?

Reply:
Honestly, I only left it because `TagNamePrefix` is used elsewhere, but you're right that it probably makes more sense to just change `TagNamePrefix` to a `ReadOnlyMemory<char>`.

Reply:
Why not just `string TagNamePrefix` that we `AsMemory()` when needed?

Reply:
That would definitely be simpler. :smile: